Vagrantfile

```diff
@@ -11,12 +19,28 @@ Vagrant.configure("2") do |config|
     config.vm.provision "shell", inline: $script, privileged: false
 
     $script = <<-SCRIPT
-        echo "deb http://apt.postgresql.org/pub/repos/apt/ xenial-pgdg main" > /etc/apt/sources.list.d/pgdg.list
+        export DEBIAN_FRONTEND="noninteractive"
+
+        # optionally enable use of ng apt cacher on the host
+        export NG_PROXY_URL="#{ENV['NG_PROXY_URL']}"
+        if [ "$NG_PROXY_URL" != "" ]; then
+            echo "Enabling Apt Proxy ($NG_PROXY_URL) ..."
+            echo 'Acquire::http { Proxy "$NG_PROXY_URL"; };' > /etc/apt/apt.conf.d/02proxy
+        fi
+
+        echo "deb http://apt.postgresql.org/pub/repos/apt/ focal-pgdg main" > /etc/apt/sources.list.d/pgdg.list
         wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add -
         add-apt-repository -y ppa:deadsnakes/ppa
 
         apt-get update
-        apt-get install -y build-essential libsnappy-dev
+        apt-get install -y build-essential libsnappy-dev postgresql-common
+
+        # no point creating the default cluster as its not used for tests
+        sed -i "s/^#start_conf.*/start_conf='manual'/g" /etc/postgresql-common/createcluster.conf
+        sed -i "s/^#create_main_cluster.*/create_main_cluster=false/g" /etc/postgresql-common/createcluster.conf
+
+        apt-get install -y python3.6 python3.6-dev python3.6-venv python3.7 python3.7-dev python3.7-venv python3.8 python3.8-dev python3.8-venv
```

Comment:
Any chance to also add py 3.9?

Reply:
@jankatins The Vagrantfile already use a ppa, so ive added python 3.9 as well :-) 

Ive rebased on lastest master hopefully the flaky test was resolved with recent commits